pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java

```diff
@@ -330,4 +331,15 @@ private static AggregateCall buildAggCall(RelNode input, AggregateCall orgAggCal
         ImmutableList.of(), aggType.isInputIntermediateFormat() ? -1 : orgAggCall.filterArg, orgAggCall.distinctKeys,
         orgAggCall.collation, numGroups, input, null, null);
   }
+
+  @Nullable
+  private static List<RexNode> findImmediateProjects(RelNode relNode) {
+    relNode = PinotRuleUtils.unboxRel(relNode);
+    if (relNode instanceof Project) {
+      return ((Project) relNode).getProjects();
+    } else if (relNode instanceof Union) {
```

Comment:
Does this apply to other `RelNode` other than `Union`?

Reply:
Yeah there could be more. For instance, if we have a Filter or even a Sort, we should ideally be able to hoist a literal up to the aggregate node.

I am a bit concerned if that could have edge-cases, that's why I kept the scope of the change in this PR small.

(digression follows)

Ideally, I think we should do this step in a separate optimization rule (run before agg exchange insert) which is run only for aggregate nodes.

Clubbing this with exchange insertion makes it a bit complicated to reason about.